tunnel-server/src/auth.ts

```diff
@@ -0,0 +1,51 @@
+import { KeyObject } from 'crypto'
+import { IncomingMessage } from 'http'
+import { calculateJwkThumbprintUri, exportJWK, jwtVerify } from 'jose'
+
+export type Claims = {
+    role: string
+    exp?: number
+}
+
+function extractBasicAuth(req: IncomingMessage){
+  const authorization = req.headers['authorization'];
+  const [scheme, data] = authorization?.split(' ') ?? []
+  if (scheme !== 'Basic') {
+    return undefined
+  }
+  const basicAuth = Buffer.from(data, 'base64').toString('ascii');
+  return basicAuth?.split(':');
```

Comment:
should make sure there are exactly 2 parts

Reply:
Looking at the RFC, it seems to allow ":" in the password part. (not the user)
Updated it

Reply:
Not that it matters, due to our usage of Basic auth header.